src/test/java/com/github/jasminb/jsonapi/DeserializationTest.java

```diff
@@ -0,0 +1,86 @@
+package com.github.jasminb.jsonapi;
+
+import com.github.jasminb.jsonapi.exceptions.DocumentSerializationException;
+import com.github.jasminb.jsonapi.models.Article;
+import com.github.jasminb.jsonapi.models.Author;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class DeserializationTest {
+    private ResourceConverter converter;
+
+    @Before
+    public void setup() {
+        converter = new ResourceConverter(Article.class, Author.class);
+    }
+
+    @Test
+    public void testCyclicalRelationshipDeserialization() throws DocumentSerializationException {
+        JSONAPIDocument<Author> deserialized = converter.readDocument(getCyclicalRelationshipJson().getBytes(), Author.class);
+
+        // Get the top-level Author
+        Author topLevelAuthorResource = deserialized.get();
+        // Get the Author through the cyclical relationship
+        Author cyclicalAuthorResource = deserialized.get().getArticles().iterator().next().getAuthor();
+
+        // Top-level Author and cyclical relationship Author should be the same object
+        Assert.assertEquals(topLevelAuthorResource, cyclicalAuthorResource);
+        // Top-level Author should preserve its attributes
+        Assert.assertNotNull(topLevelAuthorResource.getFirstName());
+        Assert.assertNotNull(topLevelAuthorResource.getLastName());
+        // Cyclical relationship Author should preserve its attributes
+        Assert.assertNotNull(cyclicalAuthorResource.getFirstName());
+        Assert.assertNotNull(cyclicalAuthorResource.getLastName());
+    }
+
+    /*
+     * Mock JSON response from an API server.
+     *
+     * ResourceConverter's serialization includes all resources (as well as a duplicate of the top-level `data` resource),
+     * in the `included` section, which doesn't expose the cyclical deserializtion bug mentioned in issue #105 on GitHub
+     * (https://github.com/jasminb/jsonapi-converter/issues/105).
+     *
+     * This method builds a JSON API response manually that contains a cyclical relationship from the `data` node through
+     * its nested relationships (Author -> Article -> Author).
+     */
+    private String getCyclicalRelationshipJson() {
```

Comment:
Please use `test/resources` for inputs and load them using `IOUtils.getResource("resource-name")`.